### PR TITLE
Adds cable coils to autolathe

### PR DESCRIPTION
**Pull request by Buggy123**
This is like the simplest thing ever why did we not do this years ago?

:cl:
add: You can now build cable coils using the autolathe.
/:cl:

### CODE REVIEW

**Comment by bgobandit** [comment]
how are cable coils made of metal and glass

**Comment by Bawhoppen** [comment]
@bgobandit you can put them into the autolathe to gain metal so don't fucking ask me

**Comment by Buggy123** [comment]
How the fuck are half the things in any of the fabs/things you can put in them made of metal and glass. Like cardboard boxes.

I literally just used the materials its made of as the cost to make it, every other tool works the same way.

**Comment by Lumipharon** [comment]
Well cables in ss13 aren't insulated so they obviously have no rubber.
power monitoring consoles can read information from all the apc's on the station, so they carry information (fiber optics)
And obviously they conduct electricity, so that's the metal.

Bam, metal and glass = ss13 cable.

**Comment by Buggy123** [comment]
Fair enough, makes perfect sense if they also use the station itself as the ground wire.

**Comment by Bawhoppen** [comment]
cable coil's desc is literally says they're a superconductor so all bets are off

**Comment by Fox-McCloud** [comment]
Just keep in mind this allows for glass/metal farming.

Take a single piece off, shove it back in and you get +50 metal and +20 glass...which means each things of cable is actually 1500 metal and 600 glass.

**Comment by Lumipharon** [comment]
Does a coil of cable act the same way as a stack of metal? If so that's just a matter of having the correct material requirements.

**Comment by Buggy123** [comment]
Uh, hmm. That might be a issue actually. I'll have to check, gimme a few hours.

**Comment by RandomMarine** [comment]
I'm not going to argue realism here. But for the sake of balanced and intuitive gameplay? Cable coils are an extremely common construction resource. Autolathe prints out common construction tools and resources. It's a no-brainer.

Also, with testing, there's no infinite metal/glass exploit. The autolathe always spends as much to create wire than what's inserted, even dividing the stacks. Only issue is that it can create coil stacks up to 50, where normally they're capped at 30.

**Comment by Buggy123** [comment]
Weird... must be a issue with how it handles stacks by default. I'll take a look at it.

**Comment by Buggy123** [comment]
Also, compared to how common it is and the price of other tools, would anyone be opposed to me lowering the material cost of cable coils so they don't cost 1,500 metal and 200 glass per stack of 30?

It's way higher than every other tool.

**Comment by Bawhoppen** [comment]
it is a giant 30 meter roll of superconductor cable as opposed to a pair of wirecutters that fit in your pocket

**Comment by Buggy123** [comment]
But it's a _superconductor_, which can carry far, far higher currents in a much smaller cable than normal wires. Technically they can't carry a infinite current because reasons but still, take a look at this:

https://commons.wikimedia.org/wiki/File:CERN-cables-p1030764.jpg

The upper one is a normal cable designed to carry 12,500 amps, the bottom is a superconducting cable that carries the same current. Notice how the bottom one is smaller than you're typical USB cord. Its literally like carrying around 30m of usb cable.

**Comment by Lumipharon** [comment]
Cables should have a maximum current you can send through them without catching on fire or some shit. Being able to just plug in 4 solars and a tesla with 200 goofballs is dull.

**Comment by Buggy123** [comment]
Maybe in another PR, for now I just want to lower the material costs a bit. so each stack doesn't cost 1500 metal.

**Comment by Buggy123** [comment]
Alright, that should be everything.

**Comment by PKPenguin321** [comment]
RIP Travis

**Comment by Buggy123** [comment]
Fuck what did i do?

File: code/game/machinery/autolathe.dm [hunk 1]
```diff
@@ -164,6 +164,9 @@
 			var/multiplier = text2num(href_list["multiplier"])
 			var/max_multiplier = min(50, being_built.materials[MAT_METAL] ?round(materials.amount(MAT_METAL)/being_built.materials[MAT_METAL]):INFINITY,being_built.materials[MAT_GLASS]?round(materials.amount(MAT_GLASS)/being_built.materials[MAT_GLASS]):INFINITY)
 			var/is_stack = ispath(being_built.build_path, /obj/item/stack)
+			
+			if ispath(being_built.build_path, /obj/item/stack/cable_coil)
+				max_multiplier = min(30, being_built.materials[MAT_METAL] ?round(materials.amount(MAT_METAL)/being_built.materials[MAT_METAL]):INFINITY,being_built.materials[MAT_GLASS]?round(materials.amount(MAT_GLASS)/being_built.materials[MAT_GLASS]):INFINITY)
```
Comment (optimumtact) [inline code/game/machinery/autolathe.dm]: I would highly suggest you rewrite this line into multiple lines

**Comment by Lumipharon** [comment]
reducing the production cost won't bring it below the salvagable amount, right?

**Comment by PKPenguin321** [comment]
fun fact, before this change the only really good ways of getting more cable were creating and destroying mechs and ordering tools crates from cargo

**Comment by Bawhoppen** [comment]
don't borgs have a cable synthesizer

**Comment by Buggy123** [comment]
Correct Lumi, build and salvage cost is reduced.

File: code/game/machinery/autolathe.dm [hunk 2]
```diff
@@ -164,7 +164,10 @@
 			var/multiplier = text2num(href_list["multiplier"])
 			var/max_multiplier = min(50, being_built.materials[MAT_METAL] ?round(materials.amount(MAT_METAL)/being_built.materials[MAT_METAL]):INFINITY,being_built.materials[MAT_GLASS]?round(materials.amount(MAT_GLASS)/being_built.materials[MAT_GLASS]):INFINITY)
 			var/is_stack = ispath(being_built.build_path, /obj/item/stack)
-
+			
+			if ispath(being_built.build_path, /obj/item/stack/cable_coil)
```
Comment (optimumtact) [inline code/game/machinery/autolathe.dm]: you need to do if( ispath(blah, blah) )

**Comment by Jalleo** [comment]
This is a okay idea but if this is done we need to remove a fair amount of cable from the station.

**Comment by optimumtact** [comment]
why?

**Comment by Lumipharon** [comment]
>why

**Comment by PKPenguin321** [comment]
The delicate wire meta

**Comment by Buggy123** [comment]
You can literally get 16 coils from primary tool storage round start, there's no way this is unbalancing.

File: code/game/machinery/autolathe.dm [hunk 3]
```diff
@@ -164,7 +164,10 @@
 			var/multiplier = text2num(href_list["multiplier"])
 			var/max_multiplier = min(50, being_built.materials[MAT_METAL] ?round(materials.amount(MAT_METAL)/being_built.materials[MAT_METAL]):INFINITY,being_built.materials[MAT_GLASS]?round(materials.amount(MAT_GLASS)/being_built.materials[MAT_GLASS]):INFINITY)
 			var/is_stack = ispath(being_built.build_path, /obj/item/stack)
-
+			
+			if (ispath(being_built.build_path, /obj/item/stack/cable_coil))
+				max_multiplier = min(30, max_multiplier)
```
Comment (tkdrg) [inline code/game/machinery/autolathe.dm]: Putting a specific ispath() check onto the generic stack code isn't ideal. Could you make this a var in the /datum/design? `stack_max` or something, make it default to 50 and override it for cable coils.

**Comment by Buggy123** [comment]
There, that SHOULD let anyone add a custom design to the autolathe with a definable maximum number of stacks to be produced, easily and simply. The question is, will it actually work?

**Comment by tkdrg** [comment]
Looks good to me. Ping me once it's squashed and I'll merge it.

**Comment by Buggy123** [comment]
It's squashed, although it seems like i got the commit message wrong again.

**Comment by optimumtact** [comment]
git commit --amend 

to rewrite the commit message of your most recent commit.

**Comment by Buggy123** [comment]
Done... ish.

**Pull request merged**